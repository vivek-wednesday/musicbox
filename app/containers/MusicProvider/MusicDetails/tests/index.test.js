/**
 *
 * Tests for MusicDetails
 *
 *
 */

import React from 'react';
import { renderProvider } from '@utils/testUtils';
import {data} from "@components/StyledCard/tests/testData"
import { MusicDetailsTest as MusicDetails } from '../index';

describe('<MusicDetails /> container tests', () => {
  const sampleData = {
    results: data
  }
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render music details', () => {
    const { findByTestId } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(findByTestId('music-details')).toBeTruthy;
  });

  it('should render music details', () => {
    const { findByTestId } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(findByTestId('price')).toBeTruthy;
  });

  it('should render music details', () => {
    const { findByTestId } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(findByTestId('name')).toBeTruthy;
  });
  
  it('should render music details', () => {
    const { findByTestId } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(findByTestId('genre')).toBeTruthy;
  });

  it('should render music details', () => {
    const { findByTestId } = renderProvider(<MusicDetails musicData={sampleData} pathName={"/track/1469577808"} />);
    expect(findByTestId('rating')).toBeTruthy;
  });
});
