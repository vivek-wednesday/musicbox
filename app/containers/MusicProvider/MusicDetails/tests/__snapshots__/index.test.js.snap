// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MusicDetails /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MusicDetails__StyleDiv-sc-1vt2rzs-0 hwYYSu"
    >
      <div
        class="ant-card ant-card-bordered"
        data-testid="music-details"
        style="width: 300px;"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <h3
                class="MusicDetails__Title-sc-1vt2rzs-3 gSqjiZ"
              >
                Broken
              </h3>
            </div>
          </div>
        </div>
        <div
          class="ant-card-cover"
        >
          <img
            alt="example"
            src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/300x300bb.jpg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <h3
            class="MusicDetails__Price-sc-1vt2rzs-2 fUrdsU"
            data-testid="price"
          >
            Price: $
            1.29
          </h3>
          <p
            class="MusicDetails__Para-sc-1vt2rzs-1 kAyqlp"
            data-testid="name"
          >
            Artist Name: 
            Jack Johnson
          </p>
          <p
            class="MusicDetails__Para-sc-1vt2rzs-1 kAyqlp"
            data-testid="genre"
          >
            Genre: 
            Rock
          </p>
          <p
            class="MusicDetails__Para-sc-1vt2rzs-1 kAyqlp"
            data-testid="rating"
          >
            Rating: 
            Unrated
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
